framework/include/libtorch/controls/LibtorchNeuralNetControl.h

```diff
@@ -0,0 +1,100 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifdef LIBTORCH_ENABLED
+
+#pragma once
+
+#include "LibtorchArtificialNeuralNet.h"
+#include "Control.h"
+
+/**
+ * A time-dependent, neural network-based control of multiple input parameters.
+ * The control strategy depends on the training of the neural net, which is
+ * typically done in a trainer object in the main app. Alternatively, the user can read
+ * neural networks using two formats:
+ * 1. Torchscript format (from python)
+ * 2. Regular data format containing the parameter values of the neural net
+ */
+class LibtorchNeuralNetControl : public Control
+{
+public:
+  static InputParameters validParams();
+
+  /// Construct using input parameters
+  LibtorchNeuralNetControl(const InputParameters & parameters);
+
+  /// Execute neural network to determine the controllable parameter values
+  virtual void execute() override;
+
+  /// Get the (signal_index)-th signal of the control neural net
+  Real getSignal(const unsigned int signal_index);
+
+  /// Get the number of controls this object is computing
+  unsigned int numberOfControlSignals() { return _control_names.size(); }
+
+  /**
+   * Function responsible for loading the neural network for the controller. This function is used
+   * when copying the neural network from a main app which trains it.
+   */
+  void loadControlNeuralNet(const std::shared_ptr<Moose::LibtorchArtificialNeuralNet> & input_nn);
```

Comment:
Why not:

```suggestion
  void loadControlNeuralNet(Moose::LibtorchArtificialNeuralNet & input_nn);
```

Here's a case where you don't need a shared_ptr; the lifetime of this control is longer than the neural net